### DISCUSSION TITLE
SONOFF SNZB-02D - Binding and Reporting setting

### DISCUSSION THREAD

**Discussion by esa1966** [Koenkk/zigbee2mqtt#24229] [Devices]
Dear support,

I have SONOFF SNZB-02D devices which doesn't work as I expected.

![image](https://github.com/user-attachments/assets/8e93d20f-ddcd-4277-8145-999c96fcbe83)

This is now more than 1 year I get this issue with reporting too quickly t°/HR/Bat.

By default, the device reports every 10 sec which is something I don't understand as t°/Hr is not changing so quick in a room (also,in HA, it records only each 5 minutes or more).

I can't change the setting in reporting once the device is BIND. I can only do this just after the BIND (very quickly).

I decided to remove / force only one device + create it again.
But, no luck, impossible to have the device binding again.
I also changed the battery as the display decrease of contrast when I try a bind (despite battery is +/- 60% charged)

So, I decided to update Z2M and the dongle FW : 
![image](https://github.com/user-attachments/assets/1a769c01-9ec6-4822-8035-c3c517b4c9dd)

After that I was only able to bind one time the device.
But impossible to change the reporting interval.
So, I press again the BIND button on the device and the device disappeared from Z2M devices list.
Again, impossible to bind again.
I know that I can restart Z2M and so on, but I spend 1 day to do all sort of think (upgrade, reboot, FW,....)
The only think I don't do is to update the FW of the SONOFF device, but I don't think it is possible without a SONOFF gateway.

I also have a SNZB-02 (without display) and this one is working fine and reports info every 10 minutes as programmed.

So, I'm stuck and I must say I'll abandon using SONOFF SNZB-02D as it doesn't work for me.

In place, I uses switchbot BLE devices which are better as it work on polling each 10 minutes. For me, it is far easier.
The battery life are more than 18 months.

On my other device, which is BIND, I set the following : 
![image](https://github.com/user-attachments/assets/45e2f977-5ef4-44ab-b3dd-8dcc9e2cb56c)

But I can see in the HA LOG MQTT that info come more often than 10 minutes : 

![image](https://github.com/user-attachments/assets/da006104-73d9-4837-a361-ce4d9afe5833)

Maybe there is something I don't understand in setting.
I just want to only have reporting based on a period of 10 minutes to increase battery life.

And it useless to report each 10 sec when HA records only changed value (more than 1 hour):  

![image](https://github.com/user-attachments/assets/25ef5f6a-83ae-41ae-9043-ebfdcfca00cb)

Are other peoples using SONOFF SNZB-02D and is it working fine for you ?
BR

**Comment by esa1966** [comment]
Test a SONOFF SNZB-02 (without the display)
![image](https://github.com/user-attachments/assets/eb65bf5e-2eea-4717-b9bb-a3be6895db6a)

In fact, there is a reporting for each cluster.
![image](https://github.com/user-attachments/assets/2a3851ad-330d-4446-b63e-e666e79d6f4a)
 
As T° and Hr are transmitted together, maybe I just need to define only T° because I just need t° each 10 minutes  I removed Hr
![image](https://github.com/user-attachments/assets/7fceecb4-e736-4079-9184-eb57ba18dc08)
 
Now, I have an update every 10 minutes : 
 
![image](https://github.com/user-attachments/assets/436a0e39-f0c6-4c7e-91e4-4aa75995e8d0)

Except for an increase of 0.5°C which is reported in 30 secondes.
 

![image](https://github.com/user-attachments/assets/53b1ee6b-4d71-4fd6-aec7-a4a02e1dd9be)

Hr is no more updated and stay at the same value.

![image](https://github.com/user-attachments/assets/ac149700-bf35-4901-ada1-ae3e01a91a97)

At the end, I add  Hr as follow:  If change for 5%, it will be transmitted in the 30 sec
![image](https://github.com/user-attachments/assets/7bfca679-c2f9-4f5c-a08a-5191ac272e67)


So, every is working as expected for the SNZB-02


I'll now do the same with SNZB-02D and post the result.

**Reply by esa1966** [reply]
About Min Rep Change precision : 

**For T° : 10 = 0.1 °C**
50 means a change of 0.5 °C

**For Hr : 10 = 0.01 %**
If I set 500 for Hr, I get correct change every 5 %

![image](https://github.com/user-attachments/assets/0259ec8a-3a1a-479a-82ff-ea3a3a87ddaa)

**Comment by esa1966** [comment]
Test a SONOFF SNZB-02D (with the display)

![image](https://github.com/user-attachments/assets/39e26de5-e537-4fee-bae2-d59eb57d5e00)

Result are correct : I use different colors to show matching for the calculation : We can see a period of 10 minutes
The one with no match is maybe the Hr Period of 3600 sec
![image](https://github.com/user-attachments/assets/48b55f4a-c15f-4739-9b1d-1fe59ae4d08d)

**Comment by esa1966** [comment]
I also want to summarize button action : 

-Double click : change F° to C°
-Press 5 Sec : Pairing mode

**And how to change setting for reporting :** 
To change setting of reporting, it is necessary : 

- For SNZB-02 : press the button for 5 sec (like the pairing mode). Then press Apply to change the value in Z2M. Then, press the button on the device (only one time is enough but you can press several time each sec) to accept the new setting transmitted by Z2M.
- SNZB-02D : you don't need to press the button for 5 Sec. Just press one time to wake up the device. Then press apply to change the value in Z2M. Then press button on the device several time each sec to accept the new value transmitted.

**With SNZB-02D, don't press 5 sec because it will go in pairing mode and reset values to default.**

**The main problem I had was because the behavior of the button on the device is not identical for SNZB-02D**.
Each time I press the Button for 5 sec, the SNZB-02D was reset to default value or I lose the device from Z2M interface (it doesn't reappear).

**So, for SNZB-02D, once bind is finished, we only need to press button 1 Sec**

You also need to press the button for 1 Sec **after** Apply the setting in Z2M : this is to accept the setting (maybe a protection to avoid external change if somebody want to hack the system.)

**Comment by NicoBerlo1** [comment]
thanks for your report, I'll try that as soon as I can, I didn't know you needed to press to confirm !

**Comment by LeMad5** [comment]
Thank you for this detailed report!
Do you know if SNZB-02D has a limit for the Min Rep Change precision for T°? I have been trying different values but in all cases, it only reports 0.2° changes. I am trying to receive updates at 0,1° T° changes.
